tests/Microsoft.Bot.Builder.AI.QnA.Tests/QnAMakerCardEqualityComparer.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.Bot.Builder.AI.QnA.Tests
+{
+    public class QnAMakerCardEqualityComparer : IEqualityComparer<IActivity>
+    {
+        public bool Equals(IActivity x, IActivity y)
+        {
+            if (x == null && y == null)
+            {
+                return true;
+            }
+
+            if (x == null || y == null)
+            {
+                return false;
+            }
+
+            var activity1 = x.AsMessageActivity();
```

Comment:
If they are a non-message activity, arent activity1 and 2 null? That would make line 29 throw NullReferenceException. Please verify that is not the case or add null checks for activity1 and activity2